targets/mbed/source/jerry_run.cpp

```diff
@@ -13,38 +14,54 @@
  * limitations under the License.
  */
 
-#include "jerry-core/jerry.h"
-
 #include <stdlib.h>
 #include <stdio.h>
 
-#include "jerry_run.h"
+#include "jerry-core/jerry.h"
 #include "jerry_extapi.h"
-
+#include "jerry_run.h"
 
 static const char* fn_sys_loop_name = "sysloop";
 
-
 int js_entry (const char *source_p, const size_t source_size)
 {
   const jerry_api_char_t *jerry_src = (const jerry_api_char_t *) source_p;
   jerry_completion_code_t ret_code = JERRY_COMPLETION_CODE_OK;
   jerry_flag_t flags = JERRY_FLAG_EMPTY;
+  jerry_api_object_t *err_obj_p = NULL;
+  jerry_api_value_t err_value = jerry_api_create_void_value ();
 
   jerry_init (flags);
 
   js_register_functions ();
 
-  if (!jerry_parse (jerry_src, source_size))
+  if (!jerry_parse (jerry_src, source_size, &err_obj_p))
   {
     printf ("Error: jerry_parse failed\r\n");
     ret_code = JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION;
+    jerry_api_release_object (err_obj_p);
   }
   else
   {
     if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
     {
-      ret_code = jerry_run ();
+      ret_code = jerry_run (&err_value);
+      jerry_api_string_t *err_str_p = NULL;
+      if (err_obj_p != NULL)
```

Comment:
This condition never has to be true, because if `jerry_parse` does not fail, then `error_obj_p` is not set. I'd replace this with an assert:
`JERRY_ASSERT (error_obj_p != NULL);`